lib/s3.py

```diff
@@ -123,22 +123,24 @@ def urlopen(url: urllib.parse.ParseResult, method='GET', headers={}, data=b'') -
     return response
 
 
-def list_bucket(url: urllib.parse.ParseResult) -> ET:
+def list_bucket(url: urllib.parse.ParseResult) -> ET.Element:
     """Get the ListBucketResult as a xml.etree.ElementTree"""
     with urlopen(url) as response:
         return ET.fromstring(response.read())
 
 
-def parse_list(result: ET, *keys) -> Iterable[Iterable[str]]:
+def parse_list(result: ET.Element, *keys) -> Generator[Union[str, None, Any], None, None]:
     """For each item in the bucket, return the given keys"""
     # 'http' url is API: see https://doc.s3.amazonaws.com/2006-03-01/AmazonS3.wsdl
     xmlns = {'s3': 'http://s3.amazonaws.com/doc/2006-03-01/'}
     for child in result.findall('s3:Contents', xmlns):
-        yield (child.find(f's3:{key}', xmlns).text for key in keys)
+        yield (child.find(f's3:{key}', xmlns).text for key in keys)  # type: ignore
 
 
 def sign_url(url: urllib.parse.ParseResult, method='GET', headers=[], duration=12 * 60 * 60) -> str:
     """Returns a "pre-signed" url for the given method and headers"""
+    if url.hostname is None:
+        sys.exit(f'hostname is None for {url.path}')
```

Comment:
This is a library -- I'd rather this was an assertion.

Reply:
Is it? It has a main and handles arguments. it doesn't really seem to be a library.

Reply:
No, that's just a `__main__` for interactively testing the library. It's normally being imported, e.g. [in image-upload](https://github.com/cockpit-project/bots/blob/main/image-upload#L32). This really isn't some "you specified the wrong arguments on the CLI" kind of situation, but an internal programming error when `sign_url` was called on invalid arguments. This should trigger a backtrace, and an assertion is exactly the right tool for function input validation.